arrays/js/arrays.js

```diff
@@ -0,0 +1,95 @@
+function isArray1(a) {
+  return (a instanceof Array);
+}
+
+function isArray2(a) {
+  return Array.prototype.isPrototypeOf(a);
+}
+
+function range(a, b, c) {
+  if (a != undefined) {
+    var l = a;
+    var index = 1;
+    var array = new Array();
+    if (b != undefined && c == undefined) {
+      l = b - a;
+      index = 2;
+    } else if (b != undefined && c != undefined) {
+      l = (b - a) / c;
+      index = 3;
+    }
+    for (var i = 0; i < l; i++) {
+      array[i] = f(i);
+    }
+    return array;
+  }
+
+  function f(x) {
+    if(index == 1) {
+      return x;
+    } else if(index == 2) {
+      return x + a;
+    } else if(index == 3) {
+      return x * c + a;
+    }
+  }
+}
+
+function compact1(arr) {
```

Comment:
could you please read task one more time?
this function should compact array
was [1, 0 , false, {}, [], true, 'string'], become [1, {}, [], true, 'string']. it should remove all falsy values